spec/allocator_spec.rb

```diff
@@ -53,27 +53,173 @@
+      # This is analogous to new_allocator(12).allocate([1/3, 2/3])
+      moneys = new_allocator(12).allocate([0.1, 0.2])
+      expect(moneys[0]).to eq(Money.new(4))
+      expect(moneys[1]).to eq(Money.new(8))
+
+      # Allocations sum to .661
+      moneys = new_allocator(488.51).allocate([0.111, 0.05, 0.5])
+      expect(moneys[0]).to eq(Money.new(82.04)) # <-- leftover penny
+      expect(moneys[1]).to eq(Money.new(36.95))
+      expect(moneys[2]).to eq(Money.new(369.52))
+
+      moneys = new_allocator(488.51).allocate([0.111, 0.05, 0.5], :roundrobin_reverse)
+      expect(moneys[0]).to eq(Money.new(82.03))
+      expect(moneys[1]).to eq(Money.new(36.95))
+      expect(moneys[2]).to eq(Money.new(369.53)) # <-- leftover penny
+
+      moneys = new_allocator(488.51).allocate([0.05, 0.111, 0.5], :nearest)
+      expect(moneys[0]).to eq(Money.new(36.95))
+      expect(moneys[1]).to eq(Money.new(82.04)) # <-- leftover penny
+      expect(moneys[2]).to eq(Money.new(369.52))
+    end
+
     specify "#allocate fills pennies from beginning to end with roundrobin strategy" do
-      moneys = new_allocator(0.05).allocate([0.3,0.7], :roundrobin)
-      expect(moneys[0]).to eq(Money.new(0.02))
-      expect(moneys[1]).to eq(Money.new(0.03))
+      #round robin for 1 penny
+      moneys = new_allocator(0.03).allocate([0.5, 0.5], :roundrobin)
+      expect(moneys[0]).to eq(Money.new(0.02)) # <-- gets 1 penny
+      expect(moneys[1]).to eq(Money.new(0.01)) # <-- gets no penny
+
+      #round robin for 2 pennies
+      moneys = new_allocator(10.55).allocate([0.25, 0.5, 0.25], :roundrobin)
+      expect(moneys[0]).to eq(Money.new(2.64)) # <-- gets 1 penny
+      expect(moneys[1]).to eq(Money.new(5.28)) # <-- gets 1 penny
+      expect(moneys[2]).to eq(Money.new(2.63)) # <-- gets no penny
+
+      #round robin for 3 pennies
+      moneys = new_allocator(195.35).allocate([0.025, 0.025, 0.125, 0.125, 0.4, 0.3], :roundrobin)
+      expect(moneys[0]).to eq(Money.new(4.89)) # <-- gets 1 penny
+      expect(moneys[1]).to eq(Money.new(4.89)) # <-- gets 1 penny
+      expect(moneys[2]).to eq(Money.new(24.42)) # <-- gets 1 penny
+      expect(moneys[3]).to eq(Money.new(24.41)) # <-- gets no penny
+      expect(moneys[4]).to eq(Money.new(78.14)) # <-- gets no penny
+      expect(moneys[5]).to eq(Money.new(58.60)) # <-- gets no penny
+
+      #round robin for 0 pennies
+      moneys = new_allocator(101).allocate([0.25, 0.25, 0.25, 0.25], :roundrobin)
+      expect(moneys[0]).to eq(Money.new(25.25)) # <-- gets no penny
+      expect(moneys[1]).to eq(Money.new(25.25)) # <-- gets no penny
+      expect(moneys[2]).to eq(Money.new(25.25)) # <-- gets no penny
+      expect(moneys[3]).to eq(Money.new(25.25)) # <-- gets no penny
     end
 
     specify "#allocate fills pennies from end to beginning with roundrobin_reverse strategy" do
+      #round robin reverse for 1 penny
       moneys = new_allocator(0.05).allocate([0.3,0.7], :roundrobin_reverse)
```

Comment:
Sidenote: I am kinda confused why the gem supports both. I probably just don't have context. But it'd be simpler for gem consumers to just reverse their array, wouldn't it?

```rb
      moneys = new_allocator(0.05).allocate([0.3,0.7].reverse)
```

Then the gem has one less thing to support.

Reply:
What do you think @elfassy ?

Reply:
my memory is a bit blurry, but I think there was a valid reason. Would need to have a closer look (on monday). Here's the original PR https://github.com/Shopify/money/pull/186